presto-druid/src/main/java/com/facebook/presto/druid/zip/ZipUtil.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.facebook.presto.druid.zip;
+
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.zip.ZipException;
+
+/**
+ * A utility class for reading and writing {@link ZipFileEntry}s from byte arrays.
+ */
+public class ZipUtil
+{
+    private ZipUtil()
+    {
+    }
+
+    /**
+     * Midnight Jan 1st 1980. Uses the current time zone as the DOS format does not support time zones
+     * and will always assume the current zone.
+     */
+    public static final long DOS_EPOCH = new GregorianCalendar(1980, Calendar.JANUARY, 1, 0, 0, 0).getTimeInMillis();
+
+    /**
+     * 23:59:59 Dec 31st 2107. The maximum date representable in DOS format.
+     */
+    public static final long MAX_DOS_DATE = new GregorianCalendar(2107, Calendar.DECEMBER, 31, 23, 59, 59).getTimeInMillis();
+
+    /* DOS format timestamp field offsets. */
+    private static final int DOS_MINUTE_OFFSET = 5;
+    private static final int DOS_HOUR_OFFSET = 11;
+    private static final int DOS_DAY_OFFSET = 16;
+    private static final int DOS_MONTH_OFFSET = 21;
+    private static final int DOS_YEAR_OFFSET = 25;
+
+    private static byte[] integerToLittleEndian(byte[] buf, int offset, long value, int numBytes)
+    {
+        for (int i = 0; i < numBytes; i++) {
+            buf[i + offset] = (byte) ((value & (0xffL << (i * 8))) >> (i * 8));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).